src/tracked_struct.rs

```diff
@@ -568,7 +568,7 @@ where
             }
         }
         if current_deps.durability < data.durability {
-            data.revisions = C::new_revisions(current_revision);
+            data.revisions = C::new_revisions(current_deps.changed_at);
             data.created_at = current_revision;
```

Comment:
It might be a bit surprising that `crated_at` can now be larger than `changed_at`

Reply:
That is already the case when we allocate the struct